modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/item/selector/ObjectEntryItemSelectorView.java

```diff
@@ -263,6 +265,14 @@ public ItemSelectorReturnType getItemSelectorReturnType() {
 					_getGroupId(), _objectDefinition.getObjectDefinitionId(),
 					searchContainer.getStart(), searchContainer.getEnd());
 
+			Stream<ObjectEntry> objectEntryStream = objectEntries.stream();
```

Comment:
Hey @mlan0 , as we discussed, the entries should be fetched using this filter directly from database. 
Please, don't forget that using Layout screen, the behavior of this same Item Selector must not be changed.